dvc/fs/fsspec_wrapper.py

```diff
@@ -93,24 +91,29 @@ def copy(self, from_info, to_info):
     def exists(self, path_info, use_dvcignore=False):
         return self.fs.exists(self._with_bucket(path_info))
 
-    def ls(
-        self, path_info, detail=False, recursive=False
-    ):  # pylint: disable=arguments-differ
-        if self.isdir(path_info) and self.is_empty(path_info):
-            return None
+    def ls(self, path_info, detail=False):
+        path = self._with_bucket(path_info)
+        files = self.fs.ls(path, detail=detail)
+        yield from self._strip_buckets(files, detail=detail)
 
+    def find(self, path_info, detail=False):
         path = self._with_bucket(path_info)
-        if recursive:
-            for root, _, files in self.fs.walk(path, detail=detail):
-                if detail:
-                    files = files.values()
-                yield from self._strip_buckets(files, detail, prefix=root)
+        files = self.fs.find(path, detail=detail)
+        if detail:
+            files = files.values()
+
+        # When calling find() on a file, it returns the same file in a list.
+        # For object-based storages, the same behavior applies to empty
+        # directories since they are represented as files. This condition
+        # checks whether we should yield an empty list (if it is an empty
+        # directory) or just yield the file itself.
+        if len(files) == 1 and files[0] == path and self.isdir(path_info):
```

Comment:
would calling full `isdir` be wasteful? Or does fsspec caching compensate for that?

Reply:
No, not really. We only call the `isdir()` if it is a file and see whether the file is actually a file or just a directory. So unless we start calling `walk_files()` on normal files, then this would cost nothing. 

The last check will simply ensure that the this is indeed a directory, and would only cost stuff when `import-url`in empty directories etc. Which I guess shouldn't be much of a big deal.